trackintel/model/util.py

```diff
@@ -90,7 +90,7 @@ def get_speed_triplegs(triplegs, positionfixes=None, method="tpls_speed"):
         grouped_pfs = positionfixes.groupby("tripleg_id").apply(_single_tripleg_mean_speed)
         # add the speed values to the triplegs column
         tpls = pd.merge(triplegs, grouped_pfs.rename("speed"), left_index=True, right_index=True)
-        tpls.index = tpls.index.astype("int64")
+        tpls.index = triplegs.index
```

Comment:
This could lead to problems if there are tripleg IDs which don't appear in the positionfixes (pd.merge would do an inner join and the number of rows would therefore not correspond). However, this can't happen because the triplegs are generated from the positionfixes, right? To be save, we could add `how="left"` in pd.merge.

Reply:
Yes I agree, I added `how="left"`, and now the method picks the left index by itself.